File: Programmers/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int number, int limit, int power) {
    int answer = 0;
    vector<int> count(number + 1);
    
    for (int i = 1; i <= number; i++)
    {
        for (int j = 1; j <= number / i; j++)
        {
            count[i * j]++;
        }
    }
    
    for (int i = 1; i <= number; i++)
    {
        if (count[i] > limit)
        {
            answer += power;
        } 
        else
        {
            answer += count[i];
        }
    }
    
    return answer;
}

File: Programmers/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int countDivisor(int n) {
        int count = 0;
        
        for (int i = 1; i * i <= n; i++) {
            if (i * i == n) {
                count++;
            } else if (n % i == 0) {
                count += 2;
            }
        }
        
        return count;
    }
    
    
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        for (int n = 1; n <= number; n++) {
            int cnt = countDivisor(n);
            
            if (cnt > limit) {
                answer += power;
                continue;
            }
            
            answer += cnt;
        }
                
        return answer;
    }
}